pkg/apis/v1alpha1/nsxserviceaccount_types.go

```diff
@@ -0,0 +1,86 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NSXServiceAccountSpec defines the desired state of NSXServiceAccount
+type NSXServiceAccountSpec struct {
+	VPCName string `json:"vpcName,omitempty"`
+}
+
+type NSXProxyEndpointAddress struct {
+	Hostname string `json:"hostname,omitempty"`
+	//+kubebuilder:validation:Format=ip
+	IP string `json:"ip,omitempty"`
+}
+
+type NSXProxyProtocol string
+
+const (
+	NSXProxyProtocolTCP NSXProxyProtocol = "TCP"
+)
+
+type NSXProxyEndpointPort struct {
+	Name     string           `json:"name,omitempty"`
+	Port     uint16           `json:"port,omitempty"`
```

Comment:
Per [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md), this should be `int32` with a range validator, ex.

```go
	// +omitempty
	// +kubebuilder:validation:Minimum:=1
	// +kubebuilder:validation:Maximum:=65535

	// Port specifies the port to access.
	Port int32 `json:"port,omitempty"`
```